provider/azure/instance.go

```diff
@@ -114,7 +114,7 @@ func instanceNetworkInterfaces(
 		return nil, nil
 	}
 	instanceNics := make(map[instance.Id][]network.Interface)
-	for ; nicsResult.NotDone(); err = nicsResult.NextWithContext(sdkCtx) {
+	for ; nicsResult.NotDone(); nicsResult.NextWithContext(sdkCtx) {
```

Comment:
I think we need to check this error as it appears that NextWithContext can do paging (as in make paginated API calls)

Reply:
all NextWithContext() calls should do error checking